### ISSUE TITLE
[Feature] Display Performer Image on Hover

### ISSUE THREAD

**Issue by ALonelyJuicebox** [stashapp/stash#4321]
This is a UI enhancement based on the successful implementation (thanks devs!) of https://github.com/stashapp/stash/issues/2575 _(Display Tag Image on hover)_
In the pursuit of maintaining UI consistency, I'm asking for the same thing, but for the Performers dropdown on the Edit page

![performer dropdown](https://github.com/stashapp/stash/assets/76471868/34bcf8aa-1438-4170-8997-76acb9cb9201)

Thanks!

**Comment by ALonelyJuicebox** [comment]
Ahh good call. #4227 looks great, and I would consider that a good replacement for this issue.

I actually disagree with you on the tag hover popup though. 🙂
The ability to see a larger image of the tag as well as a description is well worth separating the UI element out from the dropdown imo

**Comment by DingDongSoLong4** [comment]
As I understand, the reason for the popup is to make it easier to see what the tag "means" when this isn't entirely given by the tag name - i.e. "I'm not sure if I should I add this tag, let me check the description/image".

An inline image, and a one-line truncated description as described by https://github.com/stashapp/stash/pull/4227#issuecomment-1782364152 and #3076, imo gives just as much information, and is a lot cleaner, than a hover popup.

I think we can have both though: https://github.com/stashapp/stash/issues/3911#issuecomment-1647099184
tldr: add an inline image, and have the hover popup only be triggered by the image rather than the entire item.

**Comment by ALonelyJuicebox** [comment]
It sounds like we are on the same page in terms of the issue. Agreed that the reason is there to help educate the user on the intent of the tag.

Yeah man, like I said, when it comes to a performer, #4227 is great. 
Current ideal aspect ratio of a performer is vertical anyway (which aligns well with the intended design concept), and a performer is relatively identifiable, even with a small image. You don't need much to register a human face.
Performers don't have a bunch of textual information like there is with tag descriptions either so it all fits nicely.

I understand the intent of the design, and it looks great.

But for _Tags_ though...I think the design concept falls apart a bit.
The ideal aspect ratio for a Tag Image is more horizontal, it's not going to fit as well in an inline design. If the goal is to give the user feedback on the intent of the tag, that image is now going to be _considerably_ smaller, not to mention the now truncated tag description.

Conversely, with today's hover popup design I have plenty of room to see exactly what the tag image is as well as the full tag description. I just don't see the benefit for Tags. It seems like a net negative from a UX perspective.🤔

Anyway, this is all off topic from this GH issue at this point (and I can probably close this now tbh)-- maybe a topic worth discussion when #4227 is added and a consideration is made on whether or not it makes sense to change tag behavior as well
I appreciate the dialogue!

**Comment by DingDongSoLong4** [comment]
#4227 has been added - it's just only on develop builds and not in a release build yet. But yeah this probably belongs in a new issue, and this one can also probably be closed.

I agree that the popup is more useful, especially for tags. Having an inline image is probably more for consistency with performers at this point.
Personally I also wouldn't mind having a popup rather than an inline image for performers, but I think most would rather have the inline image. Perhaps we can add both inline images and popups everywhere, and make it configurable as to what combination of the two is shown?

**Comment by DogmaDragon** [comment]
Given the different purpose each implementation serves, it makes sense that they are different. A small image preview for the performer is enough to see it's what they want compared to the tag where the name can be unclear, especially if people use some other site tags instead of creating their own. 

I don't think uniformity will satisfy everyone here, as it depends on how people organize as much as what they are organizing. So more customization seems like the answer.
